rbc/src/main/java/com/library/rbc/model/Book.java

```diff
@@ -0,0 +1,34 @@
+package com.library.rbc.model;
+
+import com.library.rbc.model.enums.BookCategory;
+import com.library.rbc.model.enums.BookType;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.springframework.data.annotation.Id;
+import org.springframework.data.mongodb.core.mapping.Document;
+
+import java.util.List;
+
+@Document("books")
+@Data
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class Book {
+
+    @Id
+    private String id;
+    private String isbn;
+    private String title;
+    private String description;
+    private List<Author> authors;
+    private String imageUrl;
+    private Integer totalNumberOfCopies;
+    private BookType bookType;
+    private List<String> usersWhoFavourited;
+    private List<String> usersOnWaitingList;
+    private List<String> usersWhoSuggested;
+    private List<BookCategory> bookCategories;
+}
```

Comment:
After the figma update, I suggest adding `List<String> usersWhoRented` and `List<String> usersWhoReserved`.

<img width="926" alt="Screenshot 2024-09-26 at 15 19 39" src="https://github.com/user-attachments/assets/159ad2fd-b21e-4561-950f-e5c4344e4003">

Reply:
If you only include user IDs, then caching of user data will be necessary on the frontend. If we go with this approach, then the `Review` class doesn't need to duplicate user data (`UserDetails`), it can also store just the user ID.